Card/src/main/java/com/paypal/android/card/APIClientError.kt

```diff
@@ -0,0 +1,57 @@
+package com.paypal.android.card
+
+object APIClientError {
+
+    // 0. An unknown error occurred.
+    val unknownError = PayPalSDKError(
+        code = Code.UNKNOWN.ordinal,
+        errorDescription = "An unknown error occurred. Contact developer.paypal.com/support."
+    )
+
+    // 1. Error parsing HTTP response data.
+    val dataParsingError = PayPalSDKError(
+        code = Code.DATA_PARSING_ERROR.ordinal,
+        errorDescription = "An error occurred parsing HTTP response data. Contact developer.paypal.com/support."
+    )
+
+    // 2. Unknown Host from network.
+    val unknownHost = PayPalSDKError(
+        code = Code.UNKNOWN_HOST.ordinal,
+        errorDescription = "An error occurred due to an invalid HTTP response. Contact developer.paypal.com/support."
+    )
+
+    // 3. Missing HTTP response data.
+    val noResponseData = PayPalSDKError(
+        code = Code.NO_RESPONSE_DATA.ordinal,
+        errorDescription = "An error occurred due to missing HTTP response data. Contact developer.paypal.com/support."
+    )
+
+    // 4. There was an error constructing the URLRequest.
+    val invalidUrlRequest = PayPalSDKError(
+        code = Code.INVALID_URL_REQUEST.ordinal,
+        errorDescription = "An error occurred constructing an HTTP request. Contact developer.paypal.com/support."
+    )
+
+    // 5. The server's response body returned an error message.
+    val serverResponseError = PayPalSDKError(
+        code = Code.SERVER_RESPONSE_ERROR.ordinal,
+        errorDescription = "A server occurred. Contact developer.paypal.com/support."
+    )
+
+    // Error returned from HttpURLConnection while making request.
+    val httpURLConnectionError: (code: Int, description: String) -> PayPalSDKError = { code, description ->
```

Comment:
Would a function that takes in `code` and `description` as params and returns a `PayPalSDKError` object be more readable here?

Reply:
We made it this way to be consistent with the rest of the errors, but I think it is a good suggestion, what do you think @ndmunoz10?

Reply:
I'd prefer if possible to maintain a val property here since it's cheaper and It keeps consistency along with the other declared errors

Reply:
Makes sense and sounds good!